manim/mobject/types/vectorized_mobject.py

```diff
@@ -1096,6 +1260,134 @@ def get_subpaths(self) -> tuple:
         """
         return self.get_subpaths_from_points(self.points)
 
+    def get_subpath_split_indices_from_points(
+        self, points: np.ndarray, n_dims: int = 3, strip_null_end_curves: bool = False
+    ) -> np.ndarray:
+        nppcc = self.n_points_per_cubic_curve
+        starts = points[::nppcc]
+        ends = points[nppcc - 1 :: nppcc]
+        # This ensures that there are no more starts than ends.
+        # TODO: ends.shape[0] would be more efficient, but some test cases regarding
+        # Flash and ShowPassing flash expect a Python list instead of an ndarray, so
+        # ends.shape[0] breaks those test cases. Fix these inconsistencies.
+        n_curves = len(ends)
+        starts = starts[:n_curves]
+
+        # DO NOT DELETE THIS! Or else, a [[0 0]] ndarray will be returned
+        # for an empty list of points, when it should instead return [].
+        if n_curves == 0:
+            return np.empty((0, 2), dtype=int)
+        if n_curves == 1:
+            return np.array([[0, nppcc]])
+
+        if n_dims == 2:
+            is_equal = self.consider_points_equals_2d
+            is_different = self.consider_points_different_2d
+        else:
+            is_equal = self.consider_points_equals
+            is_different = self.consider_points_different
+
+        is_far = is_different(starts[1:], ends[:-1])
+        aux = np.arange(1, is_far.shape[0] + 1)[is_far]
+
+        split_indices = np.empty((aux.shape[0] + 1, 2), dtype=int)
+        split_indices[0, 0] = 0
+        split_indices[1:, 0] = aux
+        split_indices[:-1, 1] = aux
+        split_indices[-1, 1] = n_curves
+
+        if strip_null_end_curves:
+            for i in range(split_indices.shape[0]):
+                start_i, end_i = split_indices[i]
+                while (
+                    end_i > start_i + 1
+                    and is_equal(
+                        points[nppcc * (end_i - 1) : nppcc * end_i], ends[end_i - 2]
+                    ).all()
+                ):
+                    end_i -= 1
+                split_indices[i, 1] = end_i
+
+        split_indices *= self.n_points_per_cubic_curve
+
+        return split_indices
+
+    def get_subpath_split_indices(
+        self, n_dims: int = 3, strip_null_end_curves: bool = False
+    ) -> np.ndarray:
+        return self.get_subpath_split_indices_from_points(
+            self.points, n_dims, strip_null_end_curves
+        )
+
+    # Curve functions
+    def _init_curve_memory(self, sample_points: int = 10):
+        num_curves = self.get_num_curves()
+        lengths = np.array(
+            [self.get_nth_curve_length(n, sample_points) for n in range(num_curves)]
+        )
+
+        self.memory["piece_curves"] = {
+            "points": self.points,
```

Comment:
Do you really need to store another reference to `self.points` inside of `self.memory`?

Reply:
My idea was to store the previous state of `self.points`. In this way, the Bézier curve lengths would be recalculated only if the points change.

Now... "reference"? Oh my, I haven't noticed that! This should be a copy of `self.points`, not a reference! Thanks for bringing this up.

Reply:
Done! I addressed this issue: now `self.memory["piece_curves"]["points"]` actually holds the previous state of `self.points`, instead of an useless reference.
I also fixed some other details I missed earlier regarding `VMobject._update_curve_memory`.